mozilla-release/browser/components/migration/FirefoxProfileMigrator.js

```diff
@@ -54,6 +62,59 @@ function getFFFolder() {
   return ffFolder.exists() ? ffFolder : null;
 }
 
+function insertWholeBookmarkFolder(db, aId, aGuid) {
+  let query = `SELECT b.id, h.url, COALESCE(b.title, h.title) AS title, 
+    b.type, k.keyword, b.dateAdded, b.lastModified, h.favicon_id 
+    FROM moz_bookmarks b 
+    LEFT JOIN moz_places h ON b.fk = h.id 
+    LEFT JOIN moz_keywords k ON k.id = b.keyword_id 
+    WHERE b.type IN (1,2) AND b.parent = ${aId}
+    ORDER BY b.position;`;
```

Comment:
I didn't test with older version. But checked the changes of the schemes since firefox 3. They have removed the table: `moz_bookmarks_root`. Chromium uses it to load root bookmarks id, we don't. So this function could work with older database version also.